addons/im_livechat/models/discuss_channel.py

```diff
@@ -204,6 +205,21 @@ def _compute_livechat_agent_requesting_help_history(self):
                 else None
             )
 
+    @api.depends("chatbot_current_step_id", "chatbot_current_step_id.step_type", "chatbot_message_ids.user_script_answer_id")
+    def _compute_chatbot_current_step_is_last(self):
+        for channel in self:
+            if not channel.chatbot_current_step_id:
+                channel.chatbot_current_step_is_last = False
+            else:
+                # sudo: visitors can access the current step of the script
+                next_step = channel.sudo().chatbot_current_step_id._fetch_next_step(
+                    channel.chatbot_message_ids.user_script_answer_id
+                )
+                channel.chatbot_current_step_is_last = (
+                    channel.chatbot_current_step_id.step_type != "question_selection"
+                    and not next_step
+                )
+
```

Comment:
This should probably not be stored as the dependencies are too complex to get right.

`_fetch_next_step` is accessing many things that are not defined here, such as pretty much the whole setup of the chatbot script.